analyzers/src/SonarAnalyzer.Common/AnalysisContext/DiagnosticAnalyzerContextHelper.cs

```diff
@@ -91,9 +91,9 @@ internal static class DiagnosticAnalyzerContextHelper
                                                                                      Action<CodeBlockStartAnalysisContext<TLanguageKindEnum>> action) where TLanguageKindEnum : struct =>
         context.RegisterCodeBlockStartAction<TLanguageKindEnum>(c =>
             {
-                if (SonarAnalysisContext.ShouldAnalyze(context.TryGetValue, context.TryGetValue, generatedCodeRecognizer, c.GetSyntaxTree(), c.SemanticModel.Compilation, c.Options))
+                if (c.ShouldAnalyze(generatedCodeRecognizer))
```

Comment:
For me, this is very cryptic. I prefer clarity even with the downside of longer names. 

`sonarSyntaxNodeAnalysisContext` or `syntaxNodeContext` would be better in my opinion, See also the others.

Reply:
This ugly stuff will **all** get deleted in #6555, new "new" logic will live in `Execute` method with generic `context` argument